src/test/java/io/neonbee/health/HealthCheckRegistryTest.java

```diff
@@ -257,4 +276,52 @@ void testUnregister() {
         verify(mockedChecks, times(2)).register(eq(DUMMY_ID), eq(2000L), any());
         verify(mockedChecks, times(2)).unregister(eq(DUMMY_ID));
     }
+
+    @Test
+    @Timeout(value = 4, timeUnit = TimeUnit.SECONDS)
+    @DisplayName("it can request data from all health check verticles registered in shared map and consolidates the result")
+    void testCreateRequests(VertxTestContext testContext) throws Exception {
+        Checkpoint cp = testContext.checkpoint(2);
+
+        HealthCheckVerticle healthCheckVerticle1 = new HealthCheckVerticle();
+        HealthCheckVerticle healthCheckVerticle2 = new HealthCheckVerticle();
+
+        AsyncMap<String, Object> sharedMap =
+                new SharedDataAccessor(vertxMock, NeonBee.class).<String, Object>getAsyncMap(SHARED_MAP_NAME).result();
+        setValueOfPrivateField(neonBee, "sharedAsyncMap", sharedMap);
+
+        AsyncHelper.allComposite(List.of(deployVerticle(healthCheckVerticle1, vertxMock, testContext),
+                deployVerticle(healthCheckVerticle2, vertxMock, testContext))).onComplete(unused -> {
+                    new SharedDataAccessor(vertxMock, HealthCheckRegistryTest.class).getLock(SHARED_MAP_KEY)
+                            .onFailure(testContext::failNow)
+                            .compose(lock -> sharedMap.get(SHARED_MAP_KEY)
+                                    .map(namesOrNull -> namesOrNull != null ? (JsonArray) namesOrNull : new JsonArray())
+                                    .compose(qualifiedNames -> {
+                                        // register health checks in shared map
+                                        qualifiedNames.add(healthCheckVerticle1.getQualifiedName())
+                                                .add(healthCheckVerticle2.getQualifiedName());
+                                        sharedMap.put(SHARED_MAP_KEY, qualifiedNames);
+                                        cp.flag();
+                                        return succeededFuture();
+                                    }).onComplete(v -> lock.release()));
+                }).onSuccess(v -> {
+                    neonBee.getHealthCheckRegistry().createRequests()
+                            .onComplete(testContext.succeeding(result -> testContext.verify(() -> {
+                                Function<String, Long> matchingNameCount = id -> result.getJsonArray("checks").stream()
+                                        .filter(c -> ((JsonObject) c).getString("id").endsWith(id)).count();
+
+                                assertThat(matchingNameCount.apply(MemoryHealthCheck.NAME)).isEqualTo(1);
+                                // only one check each, because two health check verticles are deployed, but on the same
+                                // node -> thus the check id is the same and does not get added twice to the result.
+
+                                assertThat(result.getString("status", "A")).isEqualTo(result.getString("outcome", "B"));
```

Comment:
because I wanted to ensure that the test does not succeed if both status and outcome are null (for whatever reason). Changed it to an additional null check as this was obviously confusing